Feature: CURA-3397 profile override dialog options

LipuFei opened the pull request.

nallath commented on cura/CuraApplication.py:
```diff
@@ -334,7 +335,15 @@ def messageBox(self, title, text, informativeText = "", detailedText = "", butto
 
     def discardOrKeepProfileChanges(self):
-        self.showDiscardOrKeepProfileChanges.emit()
+        choice = Preferences.getInstance().getValue("cura/choice_on_profile_override")
```
Could you make the "Always ask" a bit more explicit?

nallath commented on resources/qml/Preferences/GeneralPage.qml:
```diff
@@ -377,6 +379,40 @@ UM.PreferencesPage
+                    id: choiceOnProfileOverrideDropDownButton
+
+                    model: [
+                        "Always ask me this",
```
These strings are now never translated. You should use the catalog so that they can be internationalised.

nallath commented on resources/qml/Preferences/GeneralPage.qml:
```diff
@@ -377,6 +379,40 @@ UM.PreferencesPage
+                width: childrenRect.width;
+                height: childrenRect.height;
+
+                text: catalog.i18nc("@info:tooltip", ".")
```
Shouldn't the tooltip actually give some helpfull information? If it only displays a dot, we might as well leave it out all together.

LipuFei commented:
@nallath Did a rebase to make the history clear.
- Added a comment in python code to make the if-else branches clear
- Added tooltip message with translation wrapper
- Added translation wrappers for drop down box options
